### PR TITLE
tests/i2c: Add i2c tests using new philip

### LINKED ISSUE #7: Add correct error code checking for i2c NACK
Sometimes we get EIO and other times it is ENXIO.  I believe it should be ENXIO for a nack on ADDR and EIO for nack on data...

Make that a todo.  It will break things though.

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
tests/i2c: Update write byte test to verify with philip
tests/i2c dist/api: Add i2c tests to verify error codes

This PR starts using philip with the tests.  The write byte check now doesn't depend on the read byte command and instead verifies with philip.  Also adds some tests to check the correct error codes too.  The error code check may point out some problems on other boards but was tested with the nucleo board.

Fixes #7

### CODE REVIEW

File: dist/robotframework/res/api_shell.keywords.txt [hunk 1]
```diff
@@ -31,6 +31,11 @@ API Get Last Result As Integer
     ${ret}=             Convert to Integer  ${RESULT['data'][0]}
     [Return]            ${ret}
 
+API Last Result Message Should Contain
+    [Documentation]     Return result of last API call as an integer
```
Comment (smlng) [inline dist/robotframework/res/api_shell.keywords.txt]: doc needs update

File: tests/periph_i2c/tests/03__periph_i2c_error_codes.robot [hunk 2]
```diff
@@ -0,0 +1,26 @@
+*** Settings ***
+Documentation       Data driven tests to verify writing to a valid register.
```
Comment (smlng) [inline tests/periph_i2c/tests/03__periph_i2c_error_codes.robot]: doc needs update

File: tests/periph_i2c/tests/03__periph_i2c_error_codes.robot [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+*** Test Cases ***
+Wrong Address Should ENXIO
+    [Documentation]                         Verify address NACK occurs
+    API Call Should Error                   I2C Read Reg  addr=42
+    API Last Result Message Should Contain  ENXIO
+
+Data NACK Should Data EIO
+    [Documentation]                         Verify data NACK occurs
+    PHiLIP.write_and_execute                i2c.mode.nack_data  1
```
Comment (smlng) [inline tests/periph_i2c/tests/03__periph_i2c_error_codes.robot]: you can omit underscores and use spaces instead, might look/read nicer and RF supports that by default

File: tests/periph_i2c/tests/03__periph_i2c_error_codes.robot [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+Force Tags          periph  i2c
+
+*** Test Cases ***
+Wrong Address Should ENXIO
+    [Documentation]                         Verify address NACK occurs
+    API Call Should Error                   I2C Read Reg  addr=42
+    API Last Result Message Should Contain  ENXIO
+
+Data NACK Should Data EIO
```
Comment (smlng) [inline tests/periph_i2c/tests/03__periph_i2c_error_codes.robot]: this test name looks weird to me

File: tests/periph_i2c/tests/03__periph_i2c_error_codes.robot [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+...                                 I2C Acquire
+Test Teardown       I2C Release
+
+Resource            periph_i2c.keywords.txt
+Resource            api_shell.keywords.txt
+
+Force Tags          periph  i2c
+
+*** Test Cases ***
+Wrong Address Should ENXIO
```
Comment (smlng) [inline tests/periph_i2c/tests/03__periph_i2c_error_codes.robot]: I think the exact error code doesn't have value/meaning to the user/reader of this test, simpler and more descriptive (to me) sounds: `Read Register From Wrong Address Should Fail`

Sounds like you can also add a similar test like `Read Bytes From Wrong Address Should Fail`?!

Btw. `wrong` sound so lazy,  maybe `invalid` would fit better in this context?

Comment (MrKevinWeiss) [reply tests/periph_i2c/tests/03__periph_i2c_error_codes.robot]: Yes to invalid but we should distiguish between error codes.  Our API states it for one, and there are practical reasons why one would need to know the difference between address NACK and data NACK.

File: tests/periph_i2c/tests/03__periph_i2c_error_codes.robot (hunk 3 above)
Comment (MrKevinWeiss) [reply tests/periph_i2c/tests/03__periph_i2c_error_codes.robot]: Yes to invalid but we should distiguish between error codes. Our API states it for one, and there are practical reasons why one would need to know the difference between address NACK and data NACK.

Comment (MrKevinWeiss) [reply tests/periph_i2c/tests/03__periph_i2c_error_codes.robot]: reply in the wrong place, should be for the last comment :disappointed:

File: dist/robotframework/res/api_shell.keywords.txt (hunk 1 above)
Comment (MrKevinWeiss) [reply dist/robotframework/res/api_shell.keywords.txt]: Done

File: tests/periph_i2c/tests/03__periph_i2c_error_codes.robot (hunk 2 above)
Comment (MrKevinWeiss) [reply tests/periph_i2c/tests/03__periph_i2c_error_codes.robot]: Done

File: tests/periph_i2c/tests/03__periph_i2c_error_codes.robot (hunk 3 above)
Comment (MrKevinWeiss) [reply tests/periph_i2c/tests/03__periph_i2c_error_codes.robot]: Done

File: tests/periph_i2c/tests/03__periph_i2c_error_codes.robot (hunk 4 above)
Comment (MrKevinWeiss) [reply tests/periph_i2c/tests/03__periph_i2c_error_codes.robot]: Tried to fix.

**Comment by MrKevinWeiss** [comment]
Done

File: tests/periph_i2c/tests/03__periph_i2c_error_codes.robot [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+*** Test Cases ***
+Invalid Address Should Fail With Address NACK
+    [Documentation]                         Verify address NACK (ENXIO) occurs
+    API Call Should Error                   I2C Read Reg  addr=42
+    API Last Result Message Should Contain  ENXIO
+
+Invalid Data Should Fail With Data NACK
+    [Documentation]                         Verify data NACK (EIO) occurs
+    PHiLIP.write and execute                i2c.mode.nack data  1
```
Comment (smlng) [inline tests/periph_i2c/tests/03__periph_i2c_error_codes.robot]: regarding the error: I think there is space missing between nack and data, should be 
```
i2c.mode.nack  data  1
```

**Comment by MrKevinWeiss** [comment]
Thanks @smlng :kissing:

**Comment by smlng** [comment]
ACK and GO

**Pull request merged**